expression/builtin_time_vec.go

```diff
@@ -1886,7 +1886,17 @@ func (b *builtinHourSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
 	}
 	defer b.bufAllocator.put(buf)
 	if err = b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
-		return err
+		result.ResizeInt64(n, false)
+		i64s := result.Int64s()
+		for i := 0; i < n; i++ {
+			res, isNull, err := b.evalInt(input.GetRow(i))
+			if err != nil {
+				return err
+			}
+			result.SetNull(i, isNull)
+			i64s[i] = res
+		}
+		return nil
```

Comment:
Should we implement this in `baseBuiltinFunc` as `baseBuiltinFunc.compatibleVecEvalXxx` and unify the code?